### PR TITLE
chore: fix receivers & check ResourceRender implements.

### PR DESCRIPTION
**Pull request by qicz**
xref: https://github.com/envoyproxy/gateway/pull/4263#discussion_r1764299062

### CODE REVIEW

File: internal/infrastructure/kubernetes/infra.go [hunk 1]
```diff
@@ -17,8 +17,16 @@ import (
 
 	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
 	"github.com/envoyproxy/gateway/internal/envoygateway/config"
+	"github.com/envoyproxy/gateway/internal/infrastructure/kubernetes/proxy"
+	"github.com/envoyproxy/gateway/internal/infrastructure/kubernetes/ratelimit"
 )
 
+//nolint:gocritic
```
Comment (zirain) [inline internal/infrastructure/kubernetes/infra.go:24]: why this's needed?

Comment (qicz) [reply internal/infrastructure/kubernetes/infra.go:24]: https://github.com/envoyproxy/gateway/actions/runs/11065043979/job/30743811476

Comment (qicz) [reply internal/infrastructure/kubernetes/infra.go]: https://github.com/envoyproxy/gateway/actions/runs/11065043979/job/30743811476 The action report but maybe it is not needed.

**Review by zirain — APPROVED** [review]

**Review by shawnh2 — APPROVED** [review]

**Pull request merged**